docarray/array/storage/pqlite/backend.py

```diff
@@ -53,11 +58,56 @@ def _init_storage(
 
         self._pqlite = PQLite(n_dim, **config)
         self._offset2ids = OffsetMapping(
-            name='docarray',
+            name='docarray_mappings',
             data_path=config['data_path'],
             in_memory=False,
         )
+        from ... import DocumentArray
+        from .... import Document
 
-        if docs is not None:
+        if _docs is None:
+            return
+        elif isinstance(
+            _docs, (DocumentArray, Sequence, Generator, Iterator, itertools.chain)
+        ):
             self.clear()
-            self.extend(docs)
+            self.extend(_docs)
+        else:
+            if isinstance(_docs, Document):
+                self.append(_docs)
+
+    def __getstate__(self):
+        self._pqlite.close()
+        state = dict(self.__dict__)
+        del state['_pqlite']
+        del state['_offset2ids']
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+
+        config = state['_config']
+        config = asdict(config)
+        n_dim = config.pop('n_dim')
+
+        from pqlite import PQLite
+        from .helper import OffsetMapping
+
+        self._pqlite = PQLite(n_dim, **config)
+        self._offset2ids = OffsetMapping(
+            name='docarray_mappings',
+            data_path=config['data_path'],
+            in_memory=False,
+        )
+
+    def _get_storage_infos(self) -> Dict:
+        storage_infos = super()._get_storage_infos()
+        return {
+            'Backend': 'PQLite (https://github.com/jina-ai/pqlite)',
+            'Distance Metric': self._pqlite.metric.name,
+            'Data Path': self._config.data_path,
+            'Serialization Protocol': self._config.serialize_config.get(
+                'protocol', 'pickle'
```

Comment:
doesn't look like a good place to put a default value
let's center all of these somewhere, or just modify the factory of serialize config

Reply:
fixed